### PR TITLE
chore(config): Removes several $CONFIG keys and deprecated APIs

### LINKED ISSUE #8524: Move non-API state out of $CONFIG
A lot of refactoring in minor releases is blocked by fear of breaking plugins that access `$CONFIG->some_prop`. E.g. "views". We should identify should-be-private state and move it from `$CONFIG` to `$_ELGG`, and formally declare that off-limits. I think this could be straightforward work with low risk of breaking core stuff.

### PR DESCRIPTION
**Pull request by mrclay**
Fixes #8524

BREAKING CHANGE:
Several internal data structures are no longer stored within the plugin-accessible config space; the removed keys are listed in docs/guides/upgrading.rst. The long-deprecated remove_blacklist() function has also been removed.

### CODE REVIEW

File: docs/guides/upgrading.rst [hunk 1]
```diff
@@ -224,14 +224,27 @@ Messages will no longer get the metadata 'msg' for newly created messages. This
...
+ - ``externals_map``
+ - ``i18n_loaded_from_cache``
+ - ``language_paths``
+ - ``pagesetupdone``
+ - ``registered_tag_metadata_names``
+ - ``simplecache_enabled_in_settings``
+ - ``translations``
+ - ``viewpath``
  - ``views``
+ - ``view_path``
```
Comment (ewinslow) [inline docs/guides/upgrading.rst]: Seems like a reasonable set to remove. I don't recall any being on @iionly's list...

**Comment by ewinslow** [comment]
Lgtm. Why `$GLOBALS`?

File: docs/guides/upgrading.rst (hunk 1 above)
Comment (mrclay) [inline docs/guides/upgrading.rst]: I figures views has some refactoring left, this clears the path more.

**Comment by mrclay** [comment]
$GLOBALS is unambiguously global, context independent, and ugly (motivation to get rid of it ASAP).

**Comment by mrclay** [comment]
Will have to wait a couple days to finish this up unless anyone wants to take it.

**Comment by ewinslow** [comment]
@mrclay Ugly is good.

**Comment by ewinslow** [comment]
@mrclay What else is there to do? Why not just commit what we have here now?

**Comment by mrclay** [comment]
Should be ready.

**Comment by ewinslow** [comment]
Cool! Now to rebase...

**Pull request merged**